qiita_ware/test/test_ebi.py

```diff
@@ -413,63 +230,128 @@ def test_generate_curl_command(self):
         fna_fp = join(self.temp_dir, 'seqs.fna')
-        self.files_to_remove.append(fna_fp)
-        # demux_fp doesn't need to be removed cause is being moved to be
-        # the ppd being created and will be removed there
         demux_fp = join(self.temp_dir, 'demux.seqs')
         with open(fna_fp, 'w') as f:
             f.write(FASTA_EXAMPLE)
         with File(demux_fp, "w") as f:
             to_hdf5(fna_fp, f)
 
-        # Adding a new PreprocessedData to study 1
-        # The 3rd parameter 1 is the PreprocessedIlluminaParams id, which is
-        # not relevant for this test
+        ppd = PreprocessedData.create(Study(1),
+                                      "preprocessed_sequence_illumina_params",
+                                      1, [(demux_fp, 6)], prep_template)
+
+        return ppd
+
+    def generate_new_prep_template_and_write_demux_files(self):
+        """Creates new prep-template/demux-file to avoid duplication of code"""
+
+        # ignoring warnings generated when adding templates
+        simplefilter("ignore")
+        # creating prep template without required EBI submission columns
+        metadata_dict = {
+            'SKD6.640190': {'center_name': 'ANL',
+                            'center_project_name': 'Test Project'},
+            'SKM6.640187': {'center_name': 'ANL',
+                            'center_project_name': 'Test Project'},
+            'SKD9.640182': {'center_name': 'ANL',
+                            'center_project_name': 'Test Project'}
+        }
+        metadata = pd.DataFrame.from_dict(metadata_dict, orient='index')
+        pt = PrepTemplate.create(metadata, Study(1), "18S")
+        ppd = self.write_demux_files(pt)
+
+        return ppd
+
+    def test_init_exceptions(self):
+        # not a valid action
+        with self.assertRaises(EBISumbissionError):
+            EBISubmission(1, 'This is not a valid action')
+
+        # already submitted so can't continue
+        with self.assertRaises(EBISumbissionError):
+            EBISubmission(1, 'ADD')
+
+        ppd = self.generate_new_prep_template_and_write_demux_files()
+        # raise error as we are missing columns
+        exp_text = ("Unrecognized investigation type: 'None'. This term is "
+                    "neither one of the official terms nor one of the "
+                    "user-defined terms in the ENA ontology.\nYou are missing "
+                    "some columns in your prep template for study #1 to have "
+                    "a valid submission #3: platform, primer, "
+                    "experiment_design_description, "
+                    "library_construction_protocol.\nThese samples from study "
+                    "#1, preprocessed data #3 and prep template #2 do not "
+                    "have a valid platform: 1.SKD6.640190, 1.SKM6.640187, "
+                    "1.SKD9.640182")
+        with self.assertRaisesRegexp(EBISumbissionError, exp_text):
```

Comment:
There doesn't seem to be a regexp in the exp_text, shouldn't this be a simple assertRaises with an extra check for the error message to be correct?